qiskit/algorithms/minimum_eigen_solvers/qaoa/qaoa_circuit.py

```diff
@@ -0,0 +1,209 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Global X phases and parameterized problem hamiltonian."""
+
+from typing import Optional, Union, cast, Set
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter, ParameterVector
+from qiskit.circuit.library import BlueprintCircuit
+from qiskit.opflow import OperatorBase, I, X, CircuitStateFn, H, CircuitOp, EvolutionFactory
+
+
+class QAOACircuit(BlueprintCircuit):
+    """Global X phases and parameterized problem hamiltonian."""
+
+    def __init__(self,
+                 cost_operator: OperatorBase,
+                 param_p: int,
+                 initial_state: Optional[QuantumCircuit] = None,
+                 mixer_operator: Optional[Union[QuantumCircuit, OperatorBase]] = None,
+                 name: str = "qaoa"):
+        """
+        Constructor, following the QAOA paper https://arxiv.org/abs/1411.4028
+
+        Args:
+            cost_operator: The operator representing the cost of the optimization problem,
+                denoted as U(B, gamma) in the original paper.
```

Comment:
You could use latex for the variables, 
```
:math:`U(B, \gamma)`
```
if you want 🙂 (just be sure to make the docstring a raw docstring, `r"""`)

Reply:
In the original paper this is U(C, gamma) not U(B.... U(B... is the mixer

Reply:
@Cryoris, @woodsp-ibm changed to latex.